File: sponge-examples/examples/standalone/standalone_camel_example/standalone_camel_example.js
/**
 * Sponge Knowledge base
 * Standalone with Camel example
 */

var RouteBuilder = org.apache.camel.builder.RouteBuilder;

var javaScriptRoute = new RouteBuilder({
    configure: function() {
        var _super_ = Java.super(javaScriptRoute);
        _super_.from("sponge:spongeEngine").routeId("spongeConsumerCamelJavaScript")
                .transform().simple("JavaScript route - Received message: \${body}")
                .to("stream:out");
    }
});

function onStartup() {
    camel.context.addRoutes(javaScriptRoute);
}

File: sponge-standalone-app/examples/standalone/standalone_camel_example/standalone_camel_example.py
"""
Sponge Knowledge Base
Standalone with Camel example
"""

from org.openksavi.sponge.camel import ScriptRouteBuilder

def onInit():
    # Variables for assertions only
    sponge.setVariable("sent", False)
    sponge.setVariable("message", None)

class CamelTrigger(Trigger):
    def onConfigure(self):
        self.withEvent("spongeEvent")
    def onRun(self, event):
        message = spring.context.getBean("spongeEngine").info + ": " + event.get("message")
        camel.emit(message)
        sponge.setVariable("sent", True)

class PythonRoute(ScriptRouteBuilder):
    def configure(self):
        self.fromS("sponge:spongeEngine").routeId("spongeConsumerCamelPython") \
                .transform().simple("Python route - Received message: ${body}") \
                .process(lambda exchange: sponge.setVariable("message", exchange.getIn().getBody())) \
                .to("stream:out")

def onStartup():
    camel.context.addRoutes(PythonRoute())
    sponge.event("spongeEvent").set("message", "Send me to Camel").send()
